packages/core/src/scuttle.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * @typedef {object} ScuttleOpts
+ * @property {boolean} enabled - Whether scuttling is enabled or not.
+ * @property {Array<string|RegExp>} exceptions - List of properties to exclude from scuttling.
+ * @property {string} scuttlerName - Name of the scuttler function to use.
+ * @property {(globalRef: object, scuttle: Function) => void} scuttlerFunc - Scuttler function to use.
+ */
```

Comment:
```suggestion
 * @property {(globalRef: GlobalRef, scuttle: ScuttleFn) => void} scuttlerFunc - Scuttler function to use.
 */
 
 /**
  * @callback ScuttleFn
  * @param {GlobalRef} globalRef
  * @returns {void}
  */
```

...if I'm understanding the code right, anyway.

Although, I should mention I don't understand this API.  AFAICT, the API asks the consumer for a `GlobalRef` and a `ScuttleFn`.  It then calls the `ScuttleFn` with the `GlobalRef`.  What if `ScuttleOpts` contained something like this instead?

```js
/**
 * @typedef ScuttleOpts
 * @property {boolean} enabled - Whether scuttling is enabled or not.
 * @property {Array<string|RegExp>} exceptions - List of properties to exclude from scuttling.
 * @property {string} scuttlerName - Name of the scuttler function to use.
 * @property {ScuttleFn} scuttle
 * @property {GlobalRef} globalRef
 */
```

Also:

- re: `exceptions` - are there any properties that _should not ever_ be excluded?
- re: `scuttlerName` - what's the purpose of this?
- re: `enabled` - why would we call the function if we don't intend to use it...?

Reply:
This seems to be a mistake, `ScuttleOpts` does not accept `ScuttleFn`, it accepts either a `bool` or `{enabled, exceptions, scuttlerName}` (not sure where `@property ScuttleFn` came from).
I now better understand the confusion [here](https://github.com/LavaMoat/LavaMoat/pull/758#discussion_r1441915767).

Reply:
fixed 4864384b1

Reply:
* re: `exceptions` - not that I can think of
* re: `scuttlerName` - see [here](https://github.com/LavaMoat/LavaMoat/pull/758#discussion_r1441915767)
* re: `enabled` - when you put it like that, you're right, evolution-wise it (kinda) [made sense](https://github.com/LavaMoat/LavaMoat/pull/462)(?). It was about leaving this experimental and opt-in explicit.